src/Microsoft.Extensions.Hosting.Abstractions/BackgroundService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Base class for implementing a long running <see cref="IHostedService"/>.
+    /// </summary>
+    public abstract class BackgroundService : IHostedService, IDisposable
+    {
+        private static Action<object> _cancelToken = CancelToken;
+
+        private Task _executingTask;
+        private CancellationTokenSource _executionCts;
```

Comment:
why initialize one CTS and not the other?

Reply:
I could do both I guess.